docs/usage.rst

```diff
@@ -47,6 +47,15 @@ which provides methods to run queries and manage transactions.
     asyncio.get_event_loop().run_until_complete(main())
 
 
+The :class:`Connection <asyncpg.connection.Connection>` can also be used as an
+asynchronous context manager, which will take care of closing the connection
+for you:
+
+-- code-block:: python
+    async with asyncpg.connect('postgresql://postgres@localhost/test') as conn:
```

Comment:
This should really be `async with await asyncpg.connect()`.   I don't think this is a particularly nice looking API.  For readability and to avoid making such mistakes, this formulation is better:

```python
conn  = await asyncpg.connect()
async with conn:
    ...
```

But the problem with the above is that the `with` block isn't really aligned with the connection lifetime, so I can also write:

```python

conn = await asyncpg.connect()
foo = await conn.fetch('SELECT...')
async with conn:
    await conn.fetch(...)
```

A nicer approach would be to make `connect()` return an async wrapper like the `CursorFactory`, so  you can do `async with connect() as conn:`  (without an additional `await`).

Reply:
+1.